netbout-web/src/main/java/com/netbout/rest/TkStart.java

```diff
@@ -67,13 +75,45 @@ public TkStart(final Base bse) {
 
     @Override
     public Response act(final Request req) throws IOException {
-        final Inbox inbox = new RqAlias(this.base, req).alias().inbox();
-        final long number = inbox.start();
+        final Href href = new RqHref.Base(req).href();
+        final Iterator<String> token = href.param("token").iterator();
+        String tokenkey = "";
+        final Inbox inbox;
+        long number = 0;
+        while (token.hasNext()) {
+            tokenkey = token.next();
+        }
+        if (tokenkey.equals("")) {
+            inbox = new RqAlias(this.base, req).alias().inbox();
+        } else {
+            if (this.tokencache.containsKey(tokenkey)) {
+                final int hoursInDay = 24;
```

Comment:
@prahladyeri let's use `TimeUnit` for these conversions. See [this](http://stackoverflow.com/a/19998563/4225089) for example.